AppKit/CPTokenField.j

```diff
@@ -528,12 +528,18 @@ CPTokenFieldDeleteButtonType     = 1;
     if (!CPTokenFieldInputDidBlur)
```

Comment:
Before this line we will try to blur the input field. Should this not be under a ```if``` statement as in the CPTextField counter part? Otherwise we should be able to remove the variable ```CPTokenFieldInputIsActive``` as it is never read?

This is very complex part and I feel that I don't really understand the whole picture of it. I don't know if this is needed or not but maybe it should be better to have the same functionality as the CPTextField counterpart.

Reply:
@mrcarlberg i do not understand either.
as nobody complained about DOM exceptions from tokenfields,  my feeling is to be on the safe side and revert ae52809.
do you agree?

Reply:
It is easy to get it to crash with a token field. That is how I found the problem.

Reply:
I think we should have the same behavior in both the text field and token field. They should work in the same way as it will be easier for us to understand what is going on when we look at this in a couple of years.

Reply:
then i suggest to add the missing if statement.

Reply:
This is complex and I really don't know why we do this if in the text field in the first place, but I think it will be safer and easier to understand to have the same functionality in both classes.

Reply:
Yes, please add the ```if```

Reply:
we really need to rework all this on the basis of a CPTextView-based fieldeditor :-)